diff --git a/Foundation/include/Poco/Platform.h b/Foundation/include/Poco/Platform.h
index 799db417e..5dfd7daae 100644
--- a/Foundation/include/Poco/Platform.h
+++ b/Foundation/include/Poco/Platform.h
@@ -135,6 +135,7 @@
 #define POCO_ARCH_AARCH64 0x0f
 #define POCO_ARCH_ARM64   0x0f // same as POCO_ARCH_AARCH64
 #define POCO_ARCH_RISCV64 0x10
+#define POCO_ARCH_LOONG64 0x11 // same as POCO_ARCH_LOONGARCH64
 
 
 #if defined(__ALPHA) || defined(__alpha) || defined(__alpha__) || defined(_M_ALPHA)
@@ -228,6 +229,9 @@
 #elif defined(__riscv) && (__riscv_xlen == 64)
 	#define POCO_ARCH POCO_ARCH_RISCV64
 	#define POCO_ARCH_LITTLE_ENDIAN 1
+#elif defined(__loongarch64)
+    #define POCO_ARCH POCO_ARCH_LOONG64
+	#define POCO_ARCH_LITTLE_ENDIAN 1
 #endif
 
 
diff --git a/Foundation/src/utils.h b/Foundation/src/utils.h
index 69cea1aca..cd4de1dc4 100644
--- a/Foundation/src/utils.h
+++ b/Foundation/src/utils.h
@@ -63,6 +63,7 @@
     defined(__SH4__) || defined(__alpha__) || \
     defined(_MIPS_ARCH_MIPS32R2) || \
     defined(__AARCH64EL__) || \
+    defined(__loongarch64) || \
     defined(nios2) || defined(__nios2) || defined(__nios2__)
 #define DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS 1
 #elif defined(_M_IX86) || defined(__i386__) || defined(__i386)
