diff --git a/cmake/find/zlib.cmake b/cmake/find/zlib.cmake
index bd96424b60d..0f453cf353e 100644
--- a/cmake/find/zlib.cmake
+++ b/cmake/find/zlib.cmake
@@ -10,6 +10,8 @@ else ()
     endif ()
 endif ()
 
+message("zlib: INTERNAL_ZLIB_NAME=${INTERNAL_ZLIB_NAME} && USE_INTERNAL_ZLIB_LIBRARY=${USE_INTERNAL_ZLIB_LIBRARY}")
+
 if(NOT EXISTS "${ClickHouse_SOURCE_DIR}/contrib/${INTERNAL_ZLIB_NAME}/zlib.h")
     if(USE_INTERNAL_ZLIB_LIBRARY)
         message(WARNING "submodule contrib/${INTERNAL_ZLIB_NAME} is missing. to fix try run: \n git submodule update --init --recursive")
@@ -20,6 +22,7 @@ if(NOT EXISTS "${ClickHouse_SOURCE_DIR}/contrib/${INTERNAL_ZLIB_NAME}/zlib.h")
 endif()
 
 if (NOT USE_INTERNAL_ZLIB_LIBRARY)
+    message ("Use system zlib library")
     find_package (ZLIB)
     if (NOT ZLIB_FOUND)
         message (${RECONFIGURE_MESSAGE_LEVEL} "Can't find system zlib library")
