diff --git a/src/IO/tests/gtest_bit_io.cpp b/src/IO/tests/gtest_bit_io.cpp
index 2302a6c6135..40cf25f0fea 100644
--- a/src/IO/tests/gtest_bit_io.cpp
+++ b/src/IO/tests/gtest_bit_io.cpp
@@ -158,7 +158,7 @@ TEST_P(BitIO, WriteAndRead)
 
         BitReader reader(data.data(), data.size());
 
-        int bitpos = 0;
+        // int bitpos = 0;
         int item = 0;
         for (const auto & bv : bits_and_vals)
         {
@@ -172,7 +172,7 @@ TEST_P(BitIO, WriteAndRead)
             ASSERT_TRUE(BinaryEqual(getBits(bv.first, bv.second), reader.readBits(bv.first)));
 
             ++item;
-            bitpos += bv.first;
+            // bitpos += bv.first;
         }
     }
 }
