diff --git a/cmake/warnings.cmake b/cmake/warnings.cmake
index a398c59e981..8cd33179e4d 100644
--- a/cmake/warnings.cmake
+++ b/cmake/warnings.cmake
@@ -27,6 +27,8 @@ if (COMPILER_CLANG)
     no_warning(vla-extension)
     no_warning(zero-length-array)
     no_warning(c11-extensions)
+    no_warning(reserved-identifier)
+    no_warning(deprecated-copy-with-dtor)
 
     add_warning(comma)
     add_warning(conditional-uninitialized)
